core/arch/arch.c

```diff
@@ -3355,45 +3355,59 @@ dr_mcontext_to_priv_mcontext(priv_mcontext_t *dst, dr_mcontext_t *src)
      * if we append to dr_mcontext_t in the future we'll need
      * to check src->size here.
      */
-    if (src->size != sizeof(dr_mcontext_t) &&
-        src->size != sizeof(dr_mcontext_t) - sizeof(dr_opmask_t))
+    if (src->size > sizeof(dr_mcontext_t))
         return false;
-    if (TESTALL(DR_MC_ALL, src->flags))
+    if (TEST(DR_MC_ALL, src->flags) &&
+        IF_X86_ELSE(src->size > offsetof(priv_mcontext_t, opmask), true)) {
```

Comment:
This should check for `src->size == sizeof(dr_mcontext_t)` and should not need any future editing with field changes.

Reply:
That works, done.